### PR TITLE
(DE-855) feat(artist): Separate wikidata extraction from linkage

### PR DESCRIPTION
**Pull request by lmontier-pass**
# DS PR

## Describe your changes

Please include a summary of the changes:

* Add artist_wikidata_dump DAG
* Move wikidata extraction from artist_linkage DAG to artist_wikidata_dump DAG

## Jira ticket number and/or notion link

JIRA-ticket_number

### Type of change

* [ ] hotfix (non-breaking change which fixes an issue)
* [ ] New model
* [ ] Training
* [ ] New features
* [ ] Bug Fix
* [ ] Code Refacto
* [ ] Performance Improvements
* [ ] Test
* [ ] CI
* [ ] Config

### Checklist before requesting a review

* [ ] I have performed a self-review of my code
* [ ] My code passes CI/CD tests
* [ ] I updated README.md
* [ ] I have updated the dag
* [ ] If my changes concern incremental table, I have altered their schema to accomodate with field's creation/deletion
* [ ] I have documented the corresponding [notion page](https://www.notion.so/passcultureapp/Team-Data-engineering-Data-science-22ab0eb5ddf34dc2a854d9f0e596e91b)
* [ ] I have made corresponding changes to the [tables documentation](https://www.notion.so/passcultureapp/Documentation-Tables-175a397a8e854ff4a55ae4f3620dbe3b)
* [ ] I have made corresponding changes to the [fields glossary](https://www.notion.so/passcultureapp/854a436a8f1541e1b6ec2a65f8bab600?v=798024ba90404b139e5a17407a3bc604)
* [ ] I will create a review on slack and ensure to specify the duration of the review task: short (<10min), medium (<30min), long (>30min)

### Added tests?

* [ ] 👍 yes
* [ ] 🙅 no, because they aren't needed
* [ ] 🙋 no, because I need help
* [ ] ⏰ no, but I created a ticket

### PR title format (except for MEP)

There is a linter on the PR title format. Please respect the following format:

<details>
<summary>(ticket) type(topic): comment</summary>

* ticket surrounded by parenthesis, with optionnaly a hyphen followed by one or more digits (e.g., -1234). The first part must be one of the following strings:
  * DA
  * DE
  * AE
  * DS
  * HF
  * BSR
  * PC

* type :
The second part to specify the type of change one of the following :
  * build
  * lint
  * ci
  * docs
  * feat
  * fix
  * perf
  * refactor
  * test
  * core
  * dbt

* topic within parenthesis: 1 word e.g., (dag)

* comment: tell us your life

examples:

* :white_check_mark: (DE-124) refactor(firebase): update source field
* :x: (DE-124) refactor (firebase): update source field **(space between type and topic)**
* :x: (DE-124) airflow(firebase): update source fiedd in DAG **(wrong type)**
* :x: (DE-124) (DE-124) refactor(firebase refacto): update source field **(topic in two words)**
* :white_check_mark: (BSR) docs(github): add PR title valid format in template

</details>

### CODE REVIEW

File: jobs/ml_jobs/artist_linkage/utils/gcs_utils.py [hunk 1]
```diff
@@ -54,3 +54,29 @@ def upload_parquet(dataframe: DataFrame, gcs_path: str) -> None:
...
+    """
+    client = storage.Client()
+    storage_path = gcs_path.replace("gs://", "")
+    bucket_name = storage_path.split("/")[0]
+    base_path = storage_path.replace(f"{bucket_name}/", "")
+    bucket = client.get_bucket(bucket_name)
+    blobs = bucket.list_blobs(prefix=base_path)
+
+    dates = sorted(
+        {blob.name.split("/")[1] for blob in blobs if len(blob.name.split("/")) > 1},
```
Comment (vbusson-pass) [inline jobs/ml_jobs/artist_linkage/utils/gcs_utils.py:74]: -1 non ?

File: orchestration/dags/jobs/ml/artist_wikidata_dump.py [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+            dag=dag,
+        )
+        health_check_task >> logging_task
+
+    with TaskGroup("vm_init") as vm_init:
+        gce_instance_start = StartGCEOperator(
+            task_id="gce_start_task",
+            instance_name=GCE_INSTANCE,
+            instance_type="{{ params.instance_type }}",
+            labels={"job_type": "ml"},
```
Comment (vbusson-pass) [inline orchestration/dags/jobs/ml/artist_wikidata_dump.py:88]: tu disais pas que l'extract from wikidata durait des plombes ? 
labels={"job_type": "long_ml"} ??

Comment (lmontier-pass) [reply orchestration/dags/jobs/ml/artist_wikidata_dump.py:88]: non non c'est rapide

File: jobs/ml_jobs/artist_linkage/utils/gcs_utils.py (hunk 1 above)
Comment (lmontier) [reply jobs/ml_jobs/artist_linkage/utils/gcs_utils.py:74]: oui bien vu ! (enfin ça marchait dans ce cas là mais ça marchera mieux avec -1)

Comment (lmontier-pass) [reply jobs/ml_jobs/artist_linkage/utils/gcs_utils.py]: en fait non, ce n'était pas -1 car le -1 pointe vers le fichier. Mais j'ai quand même refacto la fonction pour que ce soit plus clair :)

File: jobs/ml_jobs/artist_linkage/match_artists_on_wikidata.py [hunk 3]
```diff
@@ -1,7 +1,7 @@
 import pandas as pd
 import typer
 
-from utils.gcs_utils import upload_parquet
+from utils.gcs_utils import get_last_date_from_bucket, upload_parquet
```
Comment (lmontier-pass) [inline jobs/ml_jobs/artist_linkage/match_artists_on_wikidata.py:4]: /gpt

**Review by vbusson-pass — APPROVED** [review]

**Pull request merged**